src/State/FocusedElement.ts

```diff
@@ -540,65 +482,31 @@ export class FocusedElementState
             currentElement,
             undefined,
             isBackward,
-            undefined,
             true
         );
 
-        let nextElement: HTMLElement | null | undefined;
-
-        if (next) {
-            let uncontrolled = next.uncontrolled;
-
-            if (uncontrolled) {
-                const isGroupperFirst = ctx.isGroupperFirst;
-                let moveOutside = false;
-
-                if (isGroupperFirst !== undefined) {
-                    const groupper = ctx.groupper?.getElement();
-                    const mover = ctx.mover?.getElement();
-                    let moveFrom: HTMLElement | undefined;
-
-                    if (
-                        isGroupperFirst &&
-                        groupper &&
-                        uncontrolled.contains(groupper)
-                    ) {
-                        moveFrom = groupper;
-                    } else if (
-                        !isGroupperFirst &&
-                        mover &&
-                        uncontrolled.contains(mover)
-                    ) {
-                        moveFrom = mover;
-                    }
-
-                    if (moveFrom) {
-                        uncontrolled = moveFrom;
-                        moveOutside = true;
-                    }
-                }
+        const nextElement = next?.element;
 
-                if (uncontrolled && ctx.uncontrolled !== uncontrolled) {
-                    // We have met an uncontrolled area, just allow default action.
-                    DummyInputManager.moveWithPhantomDummy(
-                        this._tabster,
-                        uncontrolled,
-                        moveOutside,
-                        isBackward
-                    );
-                }
+        if (nextElement) {
+            if (ctx.uncontrolled && controlTab) {
+                // We are in uncontrolled zone. We allow whatever controls it to move
```

Comment:
Have you tested this behaviour with older FUI focus utilities like FocusZone and FocusTrapZone?

Reply:
I am currently fixing a bug exactly in this place. It was caught by TMP automation in the FocusZone scenarios. And I will of course run the FUI automation to make sure it is green. Will also add more tests to cover the caught corner cases.